### PR TITLE
Create a teardown function

### PR DESCRIPTION
**Pull request by molly-moen**
This PR adds a `teardown_pythonlab` function that does the following:
- flush system out to ensure all prints show up in the console. This is ported over from a [string in the apps directory](https://github.com/code-dot-org/code-dot-org/blob/522e87478dcb3eb51f6fcbd8ab4010ee7e0d331f/apps/src/pythonlab/pythonHelpers/patches.ts#L5).
- Changes directory back to the given home directory. This is to fix an issue where if the student switches directory during their code execution, we encounter buggy behavior, such as code updates not showing up. This is because we depend on the user starting from the `Files` directory in pyodide to load their code correctly.

I put this function in the `pythonlab_setup` package, since it's a twin to `setup_pythonlab()`. If anyone can think of a better name for this package now that it includes setup and teardown, let me know (or let me know if `pythonlab_setup` still makes sense!)

Once this is in (and I change the name of the package if we think that's valuable), I will update the python lab side to use this function rather than the hard-coded patch.

Ticket: [CT-703](https://codedotorg.atlassian.net/browse/CT-703)

### CODE REVIEW

**Review by snickell — APPROVED** [review]
Its funny, I was agonizing over an exact parallel yesterday: I was adding a `finalize` method to PyCall (a third party package), existing init module, and I was trying to decide whether to rename the module.

I couldn't find a good name for a setup+teardown joint package either, but I think:
- they are often and usually paired technically, so its not a surprise when the setup module also has the teardown function
- setup/init is a better name for finding the module than anything else

Basically: I think its great, nobody will be confused, and short of splitting them into two modules one of which is tiny (boooo), I don't think any other name will be more findable than setup.
